File: java/src/main/java/route/Action.java
package route;

import org.json.JSONObject;

public class Action {

    private Connection _connection;
    private String _route;

    /**
     * Constructor
     *
     * @param connection    class that help to request API
     */
    public Action(Connection connection) {
        this._connection = connection;
        this._route = "actions/";
    }

    /**
     * Create an action
     *
     * @param action    Action you want to create
     * @return          Response
     */
    public JSONObject create(JSONObject action) {
        if (action == null) {
            throw new IllegalArgumentException("Parameter is null");
        }

        return this._connection.request(this._route, "POST", action);
    }

    /**
     * Update an action
     *
     * @param id        Action's id
     * @param action    Action you want to update
     * @return          Response
     */
    public JSONObject update(String id, JSONObject action) {
        if (action == null) {
            throw new IllegalArgumentException("Parameter is null");
        }

        return this._connection.request(this._route + id, "PUT", action);
    }

    /**
     * Get an action
     *
     * @param id    Action's id
     * @return      Response
     */
    public JSONObject get(String id) {
        if (id == null) {
            throw new IllegalArgumentException("Parameter is null");
        }

        return this._connection.request(this._route + id, "GET");
    }

    /**
     * List all actions
     *
     * @return      Response
     */
    public JSONObject list() {
        return this._connection.request(this._route, "GET");
    }

    /**
     * Delete an action
     *
     * @param id    Action's id
     * @return      Response
     */
    public JSONObject delete(String id) {
        if (id == null) {
            throw new IllegalArgumentException("Parameter is null");
        }

        return this._connection.request(this._route + id, "DELETE");
    }

    /**
     * Validate an action
     *
     * @param actionId      Action's id
     * @param validation    Parameters for validation
     * @return              Response
     */
    public JSONObject validate(String actionId, JSONObject validation) {
        if (actionId == null || validation == null) {
            throw new IllegalArgumentException("Parameter is null");
        }

        return this._connection.request(this._route + actionId + "/validation", "POST", validation);
    }

    /**
     * Test an action
     *
     * @param actionId      Action's id
     * @param test          Parameters for test
     * @return              Response
     */
    public JSONObject test(String actionId, JSONObject test) {
        if (actionId == null || test == null) {
            throw new IllegalArgumentException("Parameter is null");
        }

        return this._connection.request(this._route + actionId + "/validation", "POST", test);
    }
}

File: csharp/Driver/Action.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using RestSharp;
using NP6Api.Models;

namespace NP6Api
{
    public class Action
    {
        /// <summary>
        /// Constructor
        /// </summary>
        /// <param name="connection">class that help to request API</param>
        public Action(Connection connection)
        {
            _connection = connection;
            _route = "actions/";
        }

        /// <summary>
        /// Create an action
        /// </summary>
        /// <param name="action">Action you want to create</param>
        /// <returns>Return NResponse with StatusCode and Body</returns>
        public NResponse Create(IActionModel action)
        {
            if (action == null)
            {
                throw new ArgumentException("Parameter is null");
            }
            
            return _connection.Request(_route, Method.POST, action);
        }

        /// <summary>
        /// Update an action
        /// </summary>
        /// <param name="id">Action's id</param>
        /// <param name="action">Action fields you want to update</param>
        /// <returns>Return NResponse with StatusCode and Body</returns>
        public NResponse Update(string id, IActionModel action)
        {
            if (id == null || action == null)
            {
                throw new ArgumentException("Parameter is null");
            }

            return _connection.Request(_route + id, Method.PUT, action);
        }

        /// <summary>
        /// Get an action
        /// </summary>
        /// <param name="id">Action's id</param>
        /// <returns>Return NResponse with StatusCode and Body</returns>
        public NResponse Get(string id)
        {
            if (id == null)
            {
                throw new ArgumentException("Parameter is null");
            }

            return _connection.Request(_route + id, Method.GET);
        }

        /// <summary>
        /// List all actions
        /// </summary>
        /// <returns>Return NResponse with StatusCode and Body</returns>
        public NResponse List()
        {
            return _connection.Request(_route, Method.GET);
        }

        /// <summary>
        /// Delete an action
        /// </summary>
        /// <param name="id">Action's id</param>
        /// <returns>Return NResponse with StatusCode and Body</returns>
        public NResponse Delete(string id)
        {
            if (id == null)
            {
                throw new ArgumentException("Parameter is null");
            }

            return _connection.Request(_route + id, Method.DELETE);
        }

        /// <summary>
        /// Validate an action
        /// </summary>
        /// <param name="id">Action's id</param>
        /// <param name="validation">Parameters for validation</param>
        /// <returns>Return NResponse with StatusCode and Body</returns>
        public NResponse Validate(string id, ValidationModel validation)
        {
            if (id == null || validation == null)
            {
                throw new ArgumentException("Parameter is null");
            }

            return _connection.Request(_route + id + "/validation", Method.POST, validation);
        }

        /// <summary>
        /// Test an action
        /// </summary>
        /// <param name="id">Action's id</param>
        /// <param name="test">Parameters for test</param>
        /// <returns>Return NResponse with StatusCode and Body</returns>
        public NResponse Test(string id, TestModel test)
        {
            if (id == null || test == null)
            {
                throw new ArgumentException("Parameter is null");
            }

            return _connection.Request(_route + id + "/validation", Method.POST, test);
        }

        private readonly Connection _connection;
        private string _route;
    }
}
